server/src/main/java/org/elasticsearch/index/translog/TranslogHeader.java

```diff
@@ -168,10 +160,8 @@ private static void tryReportOldVersionError(final Path path, final FileChannel
         // 0x3f => Lucene's magic number, so we can assume it's version 1 or later
         // 0x00 => version 0 of the translog
         final byte b1 = Channels.readFromFileChannel(channel, 0, 1)[0];
-        if (b1 == 0x3f) { // LUCENE_CODEC_HEADER_BYTE
-            throw new TranslogCorruptedException(path.toString(), "translog looks like version 1 or later, but has corrupted header");
-        } else if (b1 == 0x00) { // UNVERSIONED_TRANSLOG_HEADER_BYTE
-            throw new IllegalStateException("pre-1.4 translog found [" + path + "]");
+        if (b1 == 0x3f || b1 == 0x00) { // LUCENE_CODEC_HEADER_BYTE or UNVERSIONED_TRANSLOG_HEADER_BYTE
```

Comment:
Could we just drop `tryReportOldVersionError` entirely? We throw a `TranslogCorruptedException` either way.

Reply:
Sure!